src/pages/UpdateServers.js

```diff
@@ -93,8 +102,62 @@ class UpdateServers extends BaseUpdateWizardPage {
           };
         });
       });
+    this.checkMonasca();
+  }
+
+  /**
+   * checks to see if Monasca is installed, and if it is, triggers a call to the status
+   * of each server in the model
+   */
+  checkMonasca() {
+    if(this.state.monasca === undefined) {
+      fetchJson('/api/v1/clm/monasca/is_installed')
+        .then(responseData => {
+          if(responseData.installed) {
+            getInternalModel()
+              .then((yml) => {
+                this.setState({internalModel: yml, monasca: true}, () => this.getServerStatuses());
+              }).catch((error) => {
+                console.log('error getting internal model:' + error);// eslint-disable-line no-console
+              });
+          } else {
+            this.setState({monasca: false});
+          }
+        }).catch((error) => {
+          console.log('error checking if Monasca is installed:' + error);// eslint-disable-line no-console
+        });
+    } else if(this.state.monasca === true) {
+      //if monasca is installed, get the server statuses
+      this.getServerStatuses();
+    }
+  }
+
+  getServerStatuses() {
+    if(this.state.monasca && this.state.internalModel !== undefined
+        && this.props.model !== undefined) {
+      // get the list of all servers, then load their statuses
+      // possible future enhancement: batching this
+      const serverIds = this.props.model.getIn(['inputModel','servers'])
+        .map(server => server.get('id'));
+
+      serverIds.forEach((server_id) => {
```

Comment:
This is a potentially scalability problem. At a minimum, we might want to throttle this (with `lodash.throttle`) to avoid having too many parallel back end calls at once.

Reply:
it turns out that throttle discards extra calls made during the throttle period, so I'm looking at alternative approaches... with throttle, depending on the settings I use, I get either 1 or 2 calls , while the other server requests get dropped

Reply:
There is also a promise-throttle npm library to consider. If none of these pan out via a brief investiagation, we can probably defer the scalability concern to another story.